src/crimson/os/seastore/omap_manager/btree/omap_types.h

```diff
@@ -0,0 +1,128 @@
+// -*- mode:C++; tab-width:8; c-basic-offset:2; indent-tabs-mode:t -*-
+// vim: ts=8 sw=2 smarttab
+
+#pragma once
+#include "crimson/os/seastore/seastore_types.h"
+
+namespace crimson::os::seastore::omap_manager {
+
+struct omap_node_meta_t {
+  depth_t depth = 0;
+
+  std::pair<omap_node_meta_t, omap_node_meta_t> split_into() const {
+    return std::make_pair(
+      omap_node_meta_t{depth},
+      omap_node_meta_t{depth});
+  }
+
+  static omap_node_meta_t merge_from(
+    const omap_node_meta_t &lhs, const omap_node_meta_t &rhs) {
+    assert(lhs.depth == rhs.depth);
+    return omap_node_meta_t{lhs.depth};
+  }
+
+  static std::pair<omap_node_meta_t, omap_node_meta_t>
+  rebalance(const omap_node_meta_t &lhs, const omap_node_meta_t &rhs) {
+    assert(lhs.depth == rhs.depth);
+    return std::make_pair(
+      omap_node_meta_t{lhs.depth},
+      omap_node_meta_t{lhs.depth});
+  }
+};
+
+struct omap_node_meta_le_t {
+  depth_le_t depth = init_les32(0);
+
+  omap_node_meta_le_t() = default;
+  omap_node_meta_le_t(const omap_node_meta_le_t &) = default;
+  explicit omap_node_meta_le_t(const omap_node_meta_t &val)
+    : depth(init_les32(val.depth)) {}
+
+  operator omap_node_meta_t() const {
+    return omap_node_meta_t{ depth };
+  }
+};
+
+struct omap_inner_key_t {
+  uint32_t key_off = 0;
```

Comment:
Could probably be 16 bit?

Reply:
These offset keys (omap.*_key_t) appear to be only used by merge, and internally in string_kv_node_layout.  Would it be possible to restructure the few users in omap_btree_node_impl to use iterators instead?  Then, these types would become internal to string_kv_node_layout.h and I think the resulting code would be clearer.

Reply:
It would also make it easier to reuse string_kv_node_layout.

Reply:
modified.